src/main/java/frc/robot/subsystems/swerve/ModuleIOTalonFX.java

```diff
@@ -57,56 +64,79 @@ public class ModuleIOTalonFX implements ModuleIO {
+    driveTalon = new TalonFX(driveID);
+    turnTalon = new TalonFX(steerID);
+    cancoder = new CANcoder(cancoderID);
 
     var driveConfig = new TalonFXConfiguration();
-    driveConfig.CurrentLimits.StatorCurrentLimit = 40.0;
+    // Current limits
+    // Do we want to limit supply current?
+    driveConfig.CurrentLimits.StatorCurrentLimit = Module.DRIVE_STATOR_CURRENT_LIMIT;
     driveConfig.CurrentLimits.StatorCurrentLimitEnable = true;
+    // Inverts
+    driveConfig.MotorOutput.Inverted = InvertedValue.CounterClockwise_Positive;
+    driveConfig.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+    // Sensor
+    // Meters per second
+    driveConfig.Feedback.SensorToMechanismRatio =
+        (1.0 / Module.DRIVE_GEAR_RATIO) * Module.WHEEL_RADIUS * 2 * Math.PI;
+    // Controls Gains
+    driveConfig.Slot0.kV =
+        (5800.0 * (1.0 / Module.DRIVE_GEAR_RATIO) * Module.WHEEL_RADIUS * 2 * Math.PI)
+            / 12.0; // Hypothetical based on free speed
+    driveConfig.Slot0.kA = 0.0; // TODO: Find using sysid or hand tuning
+    driveConfig.Slot0.kS = 0.0;
+    driveConfig.Slot0.kP = 0.0;
+    driveConfig.Slot0.kD = 0.0;
+
     driveTalon.getConfigurator().apply(driveConfig);
-    setDriveBrakeMode(true);
 
     var turnConfig = new TalonFXConfiguration();
-    turnConfig.CurrentLimits.StatorCurrentLimit = 30.0;
+    // Current limits
+    turnConfig.CurrentLimits.StatorCurrentLimit = Module.TURN_STATOR_CURRENT_LIMIT;
     turnConfig.CurrentLimits.StatorCurrentLimitEnable = true;
+    // Inverts
+    turnConfig.MotorOutput.Inverted =
+        IS_TURN_MOTOR_INVERTED
+            ? InvertedValue.Clockwise_Positive
+            : InvertedValue.CounterClockwise_Positive;
+    turnConfig.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+    // Fused Cancoder
+    turnConfig.Feedback.FeedbackSensorSource = FeedbackSensorSourceValue.FusedCANcoder;
+    turnConfig.Feedback.FeedbackRemoteSensorID = cancoderID;
+    turnConfig.Feedback.RotorToSensorRatio = Module.TURN_GEAR_RATIO;
+    turnConfig.Feedback.SensorToMechanismRatio = 1.0;
+    turnConfig.Feedback.FeedbackRotorOffset =
+        0.0; // Is this correct? Cancoder config should handle it
+    // Controls Gains
+    turnConfig.Slot0.kV =
+        (5800.0 / Module.TURN_GEAR_RATIO)
+            / 12.0; // Free speed over voltage, should find empirically
+    turnConfig.Slot0.kA =
+        Module.TURN_GEAR_RATIO
+            * (9.37 / 483.0)
+            / (0.004 * (12.0 / 483.0)); // Based on motor dynamics math, should find in real life
+    // gearing * Kt (torque per amp) / (intertia * resistance (nominal voltage / stall current))
+    // Yeah its messy and should be found using sysid later but its worth trying as a first guess
+    // If this works we can use a similar technique on future mechanisms
+    turnConfig.Slot0.kS = 0.0; // TODO: Find empirically
+    turnConfig.Slot0.kP = 0.0;
+    turnConfig.Slot0.kD = 0.0;
+    turnConfig.ClosedLoopGeneral.ContinuousWrap = true;
+
```

Comment:
can this be simplified in any way? some looks similar

Reply:
this is configuring the swerve drive motor than steer motor, so it looks the same since its doing motor configuration twice. We do have to keep the duplication i think since we want to configure each motor separately. Some of it will have different values once we tune the pid and ff gains, but for now they are mostly 0.